gemrb/includes/win32def.h

```diff
@@ -67,6 +67,11 @@
 #endif
 #ifndef M_PI_2
 #define M_PI_2  1.57079632679489661923 // pi/2
+#endif
+
+ /* WinAPI collision. GetObject from wingdi.h conflicts with a type used in the GUIScript source (core\ScriptEngine.h) */
```

Comment:
You can also disambiguate the call: `GemRB::GetObject`

Reply:
I think he just copied the same block from PythonHelpers.h.

Reply:
That's correct, I just used findstr to find any reference to GetObject in the source and found that workaround. It seemed to reference the same problem I was having so I went along with it. Understanding is better though, I will try your suggestion, that will probably work.

Reply:
we shouldn't have that in `PythonHelpers.h` either if we can help it

Reply:
oh, I guess since `#undef` fixes it then this is`#define`ed? Thats certainly not a symbol collision and therefore namespaces don't mean anything here.

Ok, then win32def.h is the best place to put this. Can we remove the one from the other file now?

Reply:
Have I understood this correctly? This is GUIScriptInterface.h
![intellisense](https://user-images.githubusercontent.com/1923867/83156003-c4f65400-a0f9-11ea-83fa-5b40988e8ba2.png)
That is what happens, even if I add GemRB:: before it. It's takes this GetObjectW from wingdi.h :

```
#ifdef UNICODE
#define GetObject  GetObjectW
#else
#define GetObject  GetObjectA
#endif // !UNICODE
```
Not sure how to proceed. Should I follow the trail backward to see what could be pulling in wingdi.h ?

Reply:
right, as I realized in my last comment this is _not_ a collision because MSFT decided to `#define` this; it is not an actual function. I should have read the code instead of the comment :)

still would like to just keep this contained to `win32def.h` and remove it from everywhere else.

Reply:
I'm pretty sure I only need this to be in win32def.h in the end, but I did find it elsewhere in the code. Shall I prune any other trace of this and see what happens?

Reply:
yes please :)

Reply:
Ok, the only other place this workaround is used is in PythonCallbacks.h, after removing it I still get a successful build.

I'm still confused a bit, is this just an unfortunate naming coincidence?

Reply:
This is an example of why the preprocessor is a poor tool to use for a _library_ that other projects are expected to include. They should have defined an actual function, preferably with a library prefix. They only get away with this because they are MSFT.

I'll resolve this if the other offenders have been purged.

Reply:
Aha, it looks like I am retracing the same steps as Marcel : https://github.com/gemrb/gemrb/commit/33ab531ae5fccf3e102b06afb6f537508fe87cae#diff-d956242d937fb068e328d12ecc105b95

So the fix should be ok, and just needs to be contained to win32def.h?

Reply:
Is this alright? I've rebased my local copy like so:

![I love mspaint alot](https://user-images.githubusercontent.com/1923867/83168625-a7c98180-a109-11ea-9109-8e20d941e37a.png)

Now, since I've not done this before, can I simply force push to update this PR, or will that upset the delicate balance of the force?

Reply:
You can force-push your branch, the PR won't lose anything, and you can check if everything looks in order.

Reply:
Thanks, I get nervous around git sometimes